pkg/scaffold/template/lua.go

```diff
@@ -0,0 +1,56 @@
+package template
+
+import (
+	"github.com/Masterminds/sprig/v3"
+	"github.com/imdario/mergo"
+	"github.com/pluralsh/plural/pkg/template"
+	"github.com/pluralsh/plural/pkg/utils"
+	lua "github.com/yuin/gopher-lua"
+	luar "layeh.com/gopher-luar"
+)
+
+func ExecuteLua(vals map[string]interface{}, tplate string) (map[string]interface{}, error) {
+	output := map[string]interface{}{}
+	L := lua.NewState()
+	defer L.Close()
+
+	L.SetGlobal("Var", luar.New(L, vals))
+
+	for name, function := range template.GetFuncMap() {
+		L.SetGlobal(name, luar.New(L, function))
+	}
+	for name, function := range sprig.GenericFuncMap() {
+		L.SetGlobal(name, luar.New(L, function))
+	}
+
+	if err := L.DoString(tplate); err != nil {
+		return nil, err
+	}
+	if err := utils.MapLua(L.GetGlobal("valuesYaml").(*lua.LTable), &output); err != nil {
```

Comment:
I still think the name `valuesYaml` is a bit clunky.  Feel like something more straightforward like `output` would look a bit better and be nicer for external devs to use as well in the future.